ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -439,14 +447,39 @@ private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, Ad
         if (hostBlobStore == null || !hostBlobStore.isStarted() ||
             ((hostBlobStore.getCurrentState() != ReplicaState.STANDBY) &&
                 (hostBlobStore.getCurrentState() != ReplicaState.LEADER))) {
-          hostHealthy = false;
+          serverHealthStatus = ServerHealthStatus.BAD;
           break;
         }
       }
+
+      /*
+       Checks if every disk that's assigned to a partition is also mounted and disk is healthy
+       */
+      HashSet<DiskManager> partitionsDiskManagers = new HashSet<>(storageManager.getPartitionToDiskManager().values());
+      for(Map.Entry<DiskId,DiskManager> entry : storageManager.getDiskToDiskManager().entrySet()){
+        DiskManager mountedDiskManager = entry.getValue();
+
+        //Checks Disk healthchecking is enabled and
+        // if a disk isn't healthy then store that disk to return in the response
+        if(mountedDiskManager.diskHealthCheck != null && mountedDiskManager.diskHealthCheck.enabled &&
```

Comment:
let's not do this, let's just do diskManagerConfig.enabled ...

Reply:
There is no diskManagerConfig in AmbryServerRequests. Should I just rely on mountedDiskManager.diskHealthCheck.isEnabled?

Reply:
Distributed diskmanagerconfig so we no longer rely on mountedDiskManager.diskHealthCheck.isEnabled